dashboard/lib/user_multi_auth_helper.rb

```diff
@@ -38,4 +38,25 @@ def migrate_to_multi_auth
     self.provider = 'migrated'
     save
   end
+
+  #
+  # Currently assumes the user was previously migrated using migrate_to_multi_auth.
+  #
+  def demigrate_from_multi_auth
+    return true unless migrated?
+
+    credential_type = primary_authentication_option&.credential_type
```

Comment:
It would - definitely a limitation of the current implementation.  I was thinking as a next step of writing a `clear_single_auth_fields` helper and making that part of the round-trip tests, so the round-trips would be invalid unless all fields were restored.

My question is, do we think it makes sense to merge what I've got here and do the other stuff in a follow-up PR, or should I keep going in this PR and get all those changes in together?

Reply:
👍 i think merging this PR then refactoring to use `clear_single_auth_fields` in a follow-up makes sense. it'll be nice to have demigrating at our disposal for internal testing in the meantime